app/models/user.rb

```diff
@@ -1,5 +1,17 @@
 class User < ApplicationRecord
-  has_many :posts
-  has_many :comments
-  has_many :likes
+  has_many :posts, foreign_key: 'author_id', dependent: :destroy
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  after_save :update_posts_counter
+
+  def recent_posts(limit: 3)
+    posts.order(created_at: :desc).limit(limit)
+  end
+
+  private
+
+  def update_posts_counter
+    self.posts_counter = posts.count
+  end
```

Comment:
- [ ] [optional] @zunairkhan811 Kindly consider moving this method to post model, this is a project requirement 
![image](https://github.com/zunairkhan811/blog-app/assets/61408860/0130eb82-569f-4717-8f15-bfa09d5ddf7e)

Reply:
Sure... Thanks bro! 😄

Reply:
@danmainah Hi bro! The changes asked by you has been done. 😃  Here is the link to that.. [Click to see](https://github.com/zunairkhan811/blog-app/pull/3/commits/e735201156a46755bacada8bf4dc7b4b6f19af9a)